src/main/java/org/dataloader/CacheKey.java

```diff
@@ -35,4 +35,8 @@
      * @return the cache key
      */
     Object getKey(K input);
+
+    default Object getKeyWithContext(K input, Object context) {
+        return getKey(input);
+    }
```

Comment:
This is a great PR - simple and powerful.

Can you put Javadoc on this method and I will merge it

Reply:
JavaDoc is added now.